diff --git a/src/XMonad/Core.hs b/src/XMonad/Core.hs
index 7810522..08e9120 100644
--- a/src/XMonad/Core.hs
+++ b/src/XMonad/Core.hs
@@ -682,6 +682,8 @@ recompile force = io $ do
          runProcess "ghc" ["--make"
                           , "xmonad.hs"
                           , "-i"
+                          , "-framework"
+                          , "Cocoa"
                           , "-ilib"
                           , "-fforce-recomp"
                           , "-main-is", "main"
